src/global/global_init.cc

```diff
@@ -325,6 +326,10 @@ void global_init(std::vector < const char * > *alt_def_args,
     output_ceph_version();
 
   g_ceph_context->crush_location.init_on_startup();
+
+  using Deleter = std::function<void(CephContext*)>;
+  return std::unique_ptr<CephContext, Deleter>{g_ceph_context,
+                                               [](CephContext *p) {p->put();}};
```

Comment:
since `CephContext` is a ref-counted type, i think `boost::intrusive_ptr` is a more natural fit than `std::unique_ptr` - see src/rgw/rgw_main.cc for an example